src/presenced.cpp

```diff
@@ -1323,6 +1399,13 @@ void Client::addPeer(karere::Id peer)
 
     if (isExContact(peer))
     {
+        // Notify presence of non-contact that becomes contact again
+        Presence presence = mPeersPresence[peer];
```

Comment:
Check if there's a value for the `peer`. Otherwise, do not insert it in the array. We should use:
```Presence Client::peerPresence(karere::Id peer) const```